### PR TITLE
fix(node/buffer): utf8ToBytes should return a Uint8Array

### PR DESCRIPTION
**Pull request by aapoalas**
`utf8ToBytes` was in a different file from the other `ToBytes` functions, which lead to it being missed in my earlier performance refactoring of Buffer internal APIs. It was the only one (yes, I made sure :) ) like this so I decided to move it into the same helper file as the other functions.

### CODE REVIEW

**Review by bartlomieju — COMMENTED** [review]
Is there a test that could be added or enabled in `cli/tests/node_compat/config.jsonc`?

File: ext/node/polyfills/internal_binding/_utils.ts [hunk 1]
```diff
@@ -88,6 +88,102 @@ export function utf16leToBytes(str: string, units?: number) {
   return i * 2 === length ? byteArray : byteArray.subarray(0, i * 2);
 }
 
+export function utf8ToBytes(str: string, units?: number) {
```
Comment (littledivy) [inline ext/node/polyfills/internal_binding/_utils.ts:91]: Can we use the text encoding APIs for this?

Comment (aapoalas) [reply ext/node/polyfills/internal_binding/_utils.ts:91]: I was thinking about that as well. I'm not quite sure is the answer :)

Comment (aapoalas) [reply ext/node/polyfills/internal_binding/_utils.ts]: It was indeed possible.

**Comment by aapoalas** [comment]
> Is there a test that could be added or enabled in `cli/tests/node_compat/config.jsonc`?

I tried enabling all and ended up to find that there wasn't anything more to enable, at least not anything related to this issue.

**Review by bartlomieju — APPROVED** [review]
LGTM, there's a few more tests enabled so that's great 👍 thanks for fixing this Aapo!

**Review by littledivy — APPROVED** [review]
LGTM 2

**Pull request merged**